db/migrate/20231206210501_add_confirmable_to_users.rb

```diff
@@ -0,0 +1,5 @@
+class AddConfirmableToUsers < ActiveRecord::Migration[7.1]
+  def change
+    add_column :users, :confirmed_at, :datetime
+  end
+end
```

Comment:
- [ ] Great progress on your implementation! 👏 However, it seems there's an issue with the authentication feature, preventing users from creating and logging into their accounts. When starting the project, an error (attached below) is encountered.

**Error Snapshot:**
<img width="1063" alt="Screenshot 2023-12-02 at 07 47 55" src="https://github.com/78fatemanazari/Interacti_Blog/assets/107192076/e0470bf7-b5d0-4d0e-93ae-c5fb3e9ec009">

To resolve this, let's focus on the following key points:

1. **Migrations:**
    - [ ] I noticed that you've added custom migrations to your user table for authentication. However, per the project requirements, we should be using the `devise` gem for authentication. The `devise` gem comes with prebuilt modules (`registerable`, `recoverable`, `rememberable`, `validatable`, `confirmable`, etc.) that simplify the implementation of authentication. Consider removing the custom migrations related to authentication.

2. **Devise Setup:**
    - [ ] While you've made a commendable effort to integrate `devise`, it appears you missed a crucial step: generating `devise` for your `User` model. Execute the command `rails generate devise User` to generate a migration in the `db/migrate` directory with all the necessary fields and modules for setting up `devise` authentication. For more details, refer to the [devise documentation](https://github.com/heartcombo/devise#getting-started).

Happy coding! 😊🚀